src/config/config.rs

```diff
@@ -158,15 +157,6 @@ impl Config {
         self
     }
 
-    pub fn types(mut self, types: Vec<(&str, Type)>) -> Self {
-        let mut graphql_types = BTreeMap::new();
-        for (name, type_) in types {
-            graphql_types.insert(name.to_string(), type_);
-        }
-        self.types = graphql_types;
-        self
-    }
-
```

Comment:
As receiving a `Vec` and creating a map is inefficient, I ripped this off and now we're using the version from `Setters` derive.

Reply:
We should optimize the reads for performance and writes for ergonomics. Vec is quite easy to work with in Rust, hence the API looks the way it does.

Reply:
The setter expecting a Vec it was used only for tests. Should I re-add it?

Reply:
Yes please.

Reply:
Done @tusharmath